capsul/process/xml.py

```diff
@@ -59,8 +60,11 @@ def _run_process(self):
                                      'returned by process %s' % 
                                      (len(result), len(self._function_outputs),
                                       self.id))
-                for i in xrange(len(self._function_outputs)):
-                    setattr(self, self._function_outputs[i], result[i])
+                if len(self._function_outputs) == 1:
+                    setattr(self, self._function_outputs[0], result)
```

Comment:
Actually this patch enables the code to return directly the result, in my situation a list.
Indeed, result is a list, and when _function_outputs is a one element list, the original codepicked the first element of result (_function_outputs[i], result[i] when i = 1). This patch enables to retrieve the full parameter list when a single output list is requested.